dotnetcore3.1/cookiecutter-aws-sam-quick-start-s3-dotnet/{{cookiecutter.project_name}}/test/{{cookiecutter.project_name}}.Tests/FunctionTest.cs

```diff
@@ -0,0 +1,73 @@
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+
+using Xunit;
+using Amazon.Lambda;
+using Amazon.Lambda.Core;
+using Amazon.Lambda.TestUtilities;
+using Amazon.Lambda.S3Events;
+
+using Amazon;
+using Amazon.S3;
+using Amazon.S3.Model;
+using Amazon.S3.Util;
+
+using {{cookiecutter.project_name}};
+
+namespace {{cookiecutter.project_name}}.Tests
+{
+    public class FunctionTest
+    {
+        [Fact]
+        public async Task TestS3EventLambdaFunction()
+        {
+            IAmazonS3 s3Client = new AmazonS3Client(RegionEndpoint.USWest2);
+            var context = new TestLambdaContext();
+
+            var bucketName = "lambda-S3JsonLogger-".ToLower() + DateTime.Now.Ticks;
+            var key = "text.txt";
+
+            // Create a bucket an object to setup a test data.
+            await s3Client.PutBucketAsync(bucketName);
```

Comment:
I assume this will actually create an s3 bucket everytime the test runs, can we mock these calls to run tests only locally (without creating any resource on customer's account)?

Reply:
No. It would not create a new bucket it is using the MOCK, the services are auto-mocked `using Amazon.Lambda.TestUtilities;` and `Amazon.S3.Util;`